src/osp/Active/SysNewton.cpp

```diff
@@ -28,63 +28,68 @@
 
 #include <Newton.h>
 
-using namespace osp;
-using namespace osp::active;
+using osp::active::SysNewton;
+
+using osp::active::ActiveScene;
+using osp::active::ActiveEnt;
+
+using osp::active::ACompNwtBody;
+using osp::active::ACompNwtWorld;
+using osp::active::ACompTransform;
 
 const std::string SysNewton::smc_name = "NewtonPhysics";
 
 // Callback called for every Rigid Body (even static ones) on NewtonUpdate
-void cb_force_torque(const NewtonBody* body, dFloat timestep, int threadIndex)
+void cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex)
 {
-    // Get SysNewton from Newton World
-    auto* sysNewton = static_cast<SysNewton*>(
-                NewtonWorldGetUserData(NewtonBodyGetWorld(body)));
-    ActiveScene &scene = sysNewton->get_scene();
+    // Get Scene from Newton World
+    ActiveScene* scene = static_cast<ActiveScene*>(
+                NewtonWorldGetUserData(NewtonBodyGetWorld(pBody)));
 
     // Get ACompNwtBody. Every NewtonBody created here is associated with an
     // entity that contains one.
-    auto *bodyComp = static_cast<ACompNwtBody*>(NewtonBodyGetUserData(body));
-    DataPhyRigidBody &bodyPhy = bodyComp->m_bodyData;
-
-    auto& transform = scene.get_registry().get<ACompTransform>(
-                bodyComp->m_entity);
+    auto *pBodyComp = static_cast<ACompNwtBody*>(NewtonBodyGetUserData(pBody));
+    auto &rTransformComp = scene->reg_get<ACompTransform>(pBodyComp->m_entity);
 
     // Check if transform has been set externally
-    if (transform.m_transformDirty)
+    if (rTransformComp.m_transformDirty)
     {
 
         // Set matrix
-        NewtonBodySetMatrix(body, transform.m_transform.data());
+        NewtonBodySetMatrix(pBody, rTransformComp.m_transform.data());
 
-        transform.m_transformDirty = false;
+        rTransformComp.m_transformDirty = false;
     }
 
+    // TODO: deal with changing inertia, mass or stuff
+
     // Apply force and torque
-    NewtonBodySetForce(body, bodyPhy.m_netForce.data());
-    NewtonBodySetTorque(body, bodyPhy.m_netTorque.data());
+    NewtonBodySetForce(pBody, pBodyComp->m_netForce.data());
+    NewtonBodySetTorque(pBody, pBodyComp->m_netTorque.data());
 
     // Reset accumolated net force and torque for next frame
-    bodyPhy.m_netForce = {0.0f, 0.0f, 0.0f};
-    bodyPhy.m_netTorque = {0.0f, 0.0f, 0.0f};
+    pBodyComp->m_netForce = {0.0f, 0.0f, 0.0f};
+    pBodyComp->m_netTorque = {0.0f, 0.0f, 0.0f};
 }
 
-ACompNwtBody::ACompNwtBody(ACompNwtBody&& move) :
-        m_body(move.m_body),
-        m_entity(move.m_entity),
+ACompNwtBody::ACompNwtBody(ACompNwtBody&& move) noexcept
+ : DataRigidBody(std::move(move))
+ , m_body(move.m_body)
+ , m_entity(move.m_entity)
```

Comment:
consider doing

````
, m_body(std::exchange(move.m_body, {}))
, m_body(std::exchange(movem_entity, entt::null))
```

to make the "moved-from" object less likely to be accidentally "successfully" used.